src/frontend/hooks/server/invites.ts

```diff
@@ -0,0 +1,37 @@
+import {useMutation, useSuspenseQuery} from '@tanstack/react-query';
+import {useApi} from '../../contexts/ApiContext';
+
+export const INVITE_KEY = 'invites';
+
+export function useInvites() {
+  const mapeoApi = useApi();
+  return useSuspenseQuery({
+    queryKey: [INVITE_KEY],
+    queryFn: async () => {
+      return await mapeoApi.invite.getPending();
+    },
+  });
+}
+
+export function useAcceptInvite(inviteId?: string) {
+  const mapeoApi = useApi();
+  return useMutation({
+    mutationKey: [INVITE_KEY, inviteId],
```

Comment:
react query-specific question but given that `mutationKey` doesn't seem to be used anywhere (e.g. no usage of `useMutationState()`), does it make sense to specify the `inviteId` param in the `mutationFn` instead? honestly a little bit shaky on when to use which (pass in the top-level hook vs mutationFn).

the main drawback with the current approach is that you're dealing with a potentially undefined value. not a big deal but just something i noticed when looking at usage

in this case, wondering if there's a meaningful difference between these two (there may not be!):

```js
// Whenever `invite.inviteId` changes, `mutation` changes
const mutation = useAcceptInvite(invite?.inviteId)

// the mutate function holds implicit information about the invite id to work with
// this won't do anything if `invite.inviteId` is undefined
mutation.mutate(undefined, { ... })
```

```js
// `mutation` is agnostic to invite.inviteId
const mutation = useAcceptInvite()

// the mutate function is more "functional" and doesn't hold any implicit information
// also can do a check to make sure the param of interest actually exists
if (invite?.inviteId) mutation.mutate({ inviteId: invite.inviteId }, { ... })
```

Reply:
Yea, reading though the documentation we dont need to have a query key associated with mutations.

Also it looks like the convention is follow the latter pattern, that is, let the mutation handle the params-not the top level hook

addressed: https://github.com/digidem/comapeo-mobile/pull/206/commits/09003f1bf87bbc97ac776cc15f89c4ab6f4bc4f4